src/main/java/seedu/address/model/house/Block.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.house;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a House's block number.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBlock(String)}
+ */
+public class Block {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Block numbers should only contain numbers, with or without a letter as the end. "
+                    + "There can only be at most 3 digits long.";
+    public static final String VALIDATION_REGEX = "^\\d{1,3}[a-zA-Z]?$";
+    public final String value;
+
+    /**
+     * Constructs a {@code Block}.
+     *
+     * @param block A valid block number.
+     */
+    public Block(String block) {
+        requireNonNull(block);
+        checkArgument(isValidBlock(block), MESSAGE_CONSTRAINTS);
+        value = block;
+    }
+
+    /**
+     * Returns true if a given string is a valid block number.
+     */
+    public static boolean isValidBlock(String test) {
+        return test.matches(VALIDATION_REGEX);
```

Comment:
Looks good, but have you considered that what if the user enter a block that is '0' or '00' or '00A'? I am not too sure as per our requirements can these blocks can be considered as valid because I do not think that a block in Singapore can be '0' in number. But if you done the research and there is block '0', then ignore this comment.